File: services/datastream/models/address.py
# ----------------------------------------------------------------------
# address datastream model
# ----------------------------------------------------------------------
# Copyright (C) 2007-2021 The NOC Project
# See LICENSE for details
# ----------------------------------------------------------------------

# Python modules
from typing import Optional, List

# Third-party modules
from pydantic import BaseModel

# NOC modules
from .utils import StateItem, ProjectItem


class AddressProfileItem(BaseModel):
    id: str
    name: str


class VRFItem(BaseModel):
    id: str
    name: str


class AddressDataStreamItem(BaseModel):
    id: str
    name: str
    change_id: str
    address: str
    source: str
    state: StateItem
    description: Optional[str]
    labels: Optional[List[str]]
    tags: Optional[List[str]]
    fqdn: Optional[str]
    mac: Optional[str]
    subinterface: Optional[str]
    profile: AddressProfileItem
    project: Optional[ProjectItem]
    vrf: Optional[VRFItem]

File: ui/web/ip/ipam/model/Address.js
//---------------------------------------------------------------------
// ip.ipam Address Model
//---------------------------------------------------------------------
// Copyright (C) 2007-2019 The NOC Project
// See LICENSE for details
//---------------------------------------------------------------------
console.debug("Defining NOC.ip.ipam.model.Address");

Ext.define("NOC.ip.ipam.model.Address", {
  extend: "Ext.data.Model",
  fields: [
    {
      name: "prefix",
      type: "int",
    },
    {
      name: "vrf",
      type: "int",
    },
    {
      name: "address",
      type: "string",
    },
    {
      name: "name",
      type: "string",
    },
    {
      name: "fqdn",
      type: "string",
    },
    {
      name: "mac",
      type: "string",
    },
    {
      name: "description",
      type: "string",
    },
    {
      name: "afi",
      type: "int",
    },
    {
      name: "labels",
      type: "auto",
    },
    {
      name: "tt",
      type: "string",
    },
    {
      name: "state",
      type: "string",
    },
    {
      name: "project",
      type: "string",
    },
    {
      name: "managed_object",
      type: "string",
    },
    {
      name: "subinterface",
      type: "string",
    },
    {
      name: "source",
      type: "string",
    },
    {
      name: "allocated_till",
      type: "string",
    },
    // {
    //     name: "direct_permissions",
    //     type: "auto"
    // },
    {
      name: "isFree",
      type: "boolean",
      persist: false,
    },
  ],
});
